### PR TITLE
Change def setUp to def setUpClass

### PR DESCRIPTION
**Pull request by swadeley**
Update cli testcases where setUpClass functionality is emulated within
setUp #5085 

   * I have replaced the def setUp(self) and the series of if statements with a def setUpClass(cls).
   * I have removed env1 and env2 variables (env2 variable was not used anywhere).

I ran pytest tests/foreman/cli/test_contentview.py::ContentViewTestCase

Some tests skipped or failed because I do not have a Capsule set up. Also no provisioning_server configured.

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_env_with_host_registered SKIPPED

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_delete_cv_multi_env_promoted_with_host_registered SKIPPED

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_remove_cv_version_from_multi_env_capsule_scenario FAILED

tests/foreman/cli/test_contentview.py::ContentViewTestCase::test_positive_sub_host_with_restricted_user_perm_at_default_loc FAILED

### CODE REVIEW

**Comment by pondrejk** [comment]
@swadeley there is a linter error:
```
$ flake8 .
./tests/foreman/cli/test_contentview.py:2005:47: E231 missing whitespace after ','
```
You can check this locally by running flake8, also consider getting rid of the other instances of  `# pylint: disable=unexpected-keyword-arg` in the class, those are probably no longer needed (again verifiable via flake)

**Comment by swadeley** [comment]
@pondrejk Thank you for the review. I fixed and pushed the style issue. 
I have removed the `disable=unexpected-keyword-arg` lines and am running a test locally before I commit that change.

**Comment by swadeley** [comment]
I ran the tests again on tests/foreman/cli/test_contentview.py::ContentViewTestCase

These two tests failed because I do not have CapsuleVirtualMachine or VirtualMachine as host_client:

test_positive_remove_cv_version_from_multi_env_capsule_scenario
test_positive_sub_host_with_restricted_user_perm_at_default_loc

Pushed the changes.


Thank you

File: tests/foreman/cli/test_contentview.py [hunk 1]
```diff
@@ -161,27 +159,16 @@ def _get_content_view_version_lce_names_set(content_view_id, version_id):
...
-            ContentViewTestCase.org = make_org(cached=True)
-        if ContentViewTestCase.env1 is None:
-            ContentViewTestCase.env1 = make_lifecycle_environment(
+    @classmethod
+    def setUpClass(cls):
+        """Create an organization, a life cycle environment,
+        and a product."""
+        super(ContentViewTestCase, cls).setUpClass()
+        cls.org = make_org(cached=True)
+        cls.environment = make_lifecycle_environment(
```
Comment (ldjebran) [inline tests/foreman/cli/test_contentview.py]:
```python
ls.environment = make_lifecycle_environment({u'organization-id': cls.org['id']})
```

File: tests/foreman/cli/test_contentview.py [hunk 2]
```diff
@@ -161,27 +159,14 @@ def _get_content_view_version_lce_names_set(content_view_id, version_id):
...
-        if ContentViewTestCase.org is None:
-            ContentViewTestCase.org = make_org(cached=True)
-        if ContentViewTestCase.env1 is None:
-            ContentViewTestCase.env1 = make_lifecycle_environment(
-                {u'organization-id': ContentViewTestCase.org['id']})
-        if ContentViewTestCase.env2 is None:
-            ContentViewTestCase.env2 = make_lifecycle_environment(
-                {u'organization-id': ContentViewTestCase.org['id'],
-                 u'prior': ContentViewTestCase.env1['label']})
-        if ContentViewTestCase.product is None:
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: you removed product here, but not added in setupclass, so how all statements like `repo = make_repository({'product-id': self.product['id']})` works?

File: tests/foreman/cli/test_contentview.py [hunk 3]
```diff
@@ -161,27 +159,14 @@ def _get_content_view_version_lce_names_set(content_view_id, version_id):
...
-            ContentViewTestCase.product = make_product(
-                {u'organization-id': ContentViewTestCase.org['id']},
-                cached=True)
-
-    # pylint: disable=unexpected-keyword-arg
+    @classmethod
+    def setUpClass(cls):
+        """Create an organization, a life cycle environment,
+        and a product."""
```
Comment (oshtaier) [inline tests/foreman/cli/test_contentview.py]: product?

File: tests/foreman/cli/test_contentview.py (hunk 2 above)
Comment (oshtaier) [reply tests/foreman/cli/test_contentview.py]: not a problem, please make changes, test them and ping us to merge

File: tests/foreman/cli/test_contentview.py (hunk 3 above)
Comment (swadeley) [reply tests/foreman/cli/test_contentview.py]: Hello, I have added, and pushed: `cls.product` = make_product({u'organization-id': `cls.org['id']})`

Ran tests again as per https://github.com/SatelliteQE/robottelo/pull/6450#issuecomment-437923163

**Comment by swadeley** [comment]
@ldjebran Hello, when you say "use different commits" I think you mean do not amend and force push. When I joined QE I was told  we are expected to squash commits and use force push. Do you want me to make separate commits and then squash them later?

Thank you

**Comment by swadeley** [comment]
@ldjebran Hello, when you say "use different commits" I think you mean do not amend and force push. When I joined QE I was told  we are expected to squash commits and use force push. Do you want me to make separate commits and then squash them later?

Thank you

**Comment by swadeley** [comment]
Hello

Test above failed due to lack of manifest. I set up another Satellite and Capsule in Beaker and ran the tests from our Jenkins "standalone automation" server with Pytest options set to " tests/foreman/cli/test_contentview.py::ContentViewTestCase". Only one test failed.

pytest tests/foreman/cli/test_contentview.py::ContentViewTestCase
============================= test session starts ==============================

collecting ... 2018-12-18 14:43:53 - conftest - DEBUG - Collected 88 test cases

collected 88 items / 5 deselected

=================================== FAILURES ===================================
 ContentViewTestCase.test_positive_remove_cv_version_from_multi_env_capsule_scenario 

<snip>
>       with CapsuleVirtualMachine(organization_ids=[org['id']]) as capsule_vm:

tests/foreman/cli/test_contentview.py:4246: 
<snip>
        # ensure that capsule configuration exist and validate
        if not setting_is_set('capsule'):
>           raise CapsuleVirtualMachineError('capsule configuration not set')
E           robottelo.vm_capsule.CapsuleVirtualMachineError: capsule configuration not set

============= 1 failed, 82 passed, 5 deselected in 8660.25 seconds =============

**Comment by swadeley** [comment]
Hello

Seems there is no way to use an existing Capsule for the test " ContentViewTestCase.test_positive_remove_cv_version_from_multi_env_capsule_scenario".  The test was written to call vm_capsule.py to set up a new VM on a separate "provisioning server` which I do not have locally. I am not sure why that test failed when I ran it using the Jenkins test automation server, there must be something that I have not configured.

**Comment by ldjebran** [comment]
@swadeley about different changes commits, yes the final reviewer has to squash and merge, and different commit will help us not to review all the code again but to review the changes.

**Comment by ldjebran** [comment]
@swadeley you should not worry about capsule test as not affected by your changes.

**Comment by ldjebran** [comment]
@swadeley Think @oshtaier is still waiting for the tests output , after the latest changes, just include in pytest the affected tests for the run.

**Comment by oshtaier** [comment]
No-no, I am really sorry for delay, we had:
`============= 1 failed, 82 passed, 5 deselected in 8660.25 seconds =============`
so it is good to go. Thanks

**Comment by oshtaier** [comment]
you know these holidays... =)

**Pull request merged**